probatus/feature_elimination/feature_elimination.py

```diff
@@ -354,19 +369,60 @@ def fit(self, X, y, column_names=None):
         if self.random_state is not None:
             np.random.seed(self.random_state)
 
+        # If to columns_to_keep is not provided, then initialise it by an empty string.
+        # If provided check if all the elements in columns_to_keep are of type string.
+        if columns_to_keep is None:
+            len_columns_to_keep = 0
+        else :
+            if all(isinstance(x,str) for x in columns_to_keep):
+                len_columns_to_keep = len(columns_to_keep)
+            else :
+                raise(ValueError('The current values of columns_to_keep are not allowed.All the elements should be strings.'))
+        
+        # If the column_names parameters is provided, check if the elements in both columns_to_keep & column_names are matching.
+        if column_names is not None and columns_to_keep is not None :
```

Comment:
column_names are incorporated into X when we do:
```        self.X , self.column_names = preprocess_data(X, X_name='X', column_names=column_names, verbose=self.verbose)
```
After that the X is the pandas dataframe with some column names. If column_names if None X is still going have some column names assigned. It could be that the user already passes X as df with column names in it and then column_names=None. Here i wouldn't check column_names but rather check X.column_names whether columns_to_keep are in there.

Reply:
Makes sense.Also per the docs the X should be a dataframe.

Reply:
Doesn't have to be. At the beginning of fit() we run `preprocess_data` which transforms the np.array to pd.Dataframe. That function also ensures the column names consistency